model/types.go

```diff
@@ -48,12 +48,12 @@ type Model interface {
 	GetFreeContainerMetric(FreeContainerMetricRequest) ([]statstore.TimePoint, time.Time, error)
 
 	// The GetXStats operations extract all derived stats for a single entity of the cluster.
-	GetClusterStats() (map[string]StatBundle, time.Duration, error)
-	GetNodeStats(NodeRequest) (map[string]StatBundle, time.Duration, error)
-	GetNamespaceStats(NamespaceRequest) (map[string]StatBundle, time.Duration, error)
-	GetPodStats(PodRequest) (map[string]StatBundle, time.Duration, error)
-	GetPodContainerStats(PodContainerRequest) (map[string]StatBundle, time.Duration, error)
-	GetFreeContainerStats(FreeContainerRequest) (map[string]StatBundle, time.Duration, error)
+	GetClusterStats() (map[string]StatBundle, time.Time, time.Duration, error)
```

Comment:
Also the returns types are essentially the same. Is it worth unifying them into a single struct and re-use in all methods?